File: 03_Two_Pointer/Remove_duplicates_from_sorted_Array/remove_duplicates_from_sorted_array.cpp
// LeetCode Problem 26 - https://leetcode.com/problems/remove-duplicates-from-sorted-array/description/

// Approach - 1 | One Pass | In-place | TC - O(n), SC - O(1)
    int removeDuplicates(vector<int>& nums) {
        int j = 0;
        for(int i=1;i<nums.size();i++){
            if(nums[j]!=nums[i]){
                nums[++j] = nums[i];
            }
        }
        return j+1;
    }

    // Count duplicates
    int removeDuplicates(vector<int>& nums) {
        int n = nums.size();
        int count = 0;
        for(int i = 1; i < n; i++){
            if(nums[i] == nums[i-1]) count++;
            else nums[i-count] = nums[i];
        }
        return n-count;
    }

File: 03_Two_Pointer/Remove_duplicates_from_sorted_Array/remove_duplicates_from_sorted_array.java
public class remove_duplicates_from_sorted_array {

    public int removeDuplicates(int[] nums) {
        if (nums.length == 0) return 0;

        int j = 0;  // last unique element index

        for (int i = 1; i < nums.length; i++) {
            if (nums[j] != nums[i]) {
                nums[++j] = nums[i];
            }
        }
        return j + 1;  // length of unique elements
    }

    // Counting duplicates
    public int removeDuplicates(int[] nums) {
        int n = nums.length;
        int count = 0;  // number of duplicates found

        for (int i = 1; i < n; i++) {
            if (nums[i] == nums[i - 1]) {
                count++;
            } else {
                nums[i - count] = nums[i];
            }
        }
        return n - count;  // new length
    }
}
